src/Cache/MongoLock.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace MongoDB\Laravel\Cache;
+
+use Illuminate\Cache\Lock;
+use MongoDB\Laravel\Collection;
+use MongoDB\Operation\FindOneAndUpdate;
+use Override;
+
+use function random_int;
+
+final class MongoLock extends Lock
+{
+    /**
+     * Create a new lock instance.
+     *
+     * @param  Collection  $collection              The MongoDB collection
+     * @param  string      $name
+     * @param  int         $seconds
+     * @param  string|null $owner
+     * @param  array       $lottery                 The prune probability odds
+     * @param int         $defaultTimeoutInSeconds The default number of seconds that a lock should be held
+     */
+    public function __construct(
+        private Collection $collection,
+        string $name,
+        int $seconds,
+        ?string $owner = null,
+        private array $lottery = [2, 100],
+        private int $defaultTimeoutInSeconds = 86400,
+    ) {
+        parent::__construct($name, $seconds, $owner);
+    }
+
+    /**
+     * Attempt to acquire the lock.
+     *
+     * @return bool
+     */
+    public function acquire()
+    {
+        // The lock can be acquired if: it doesn't exist, it has expired,
+        // or it is already owned by the same lock instance.
+        $condition = [
+            '$or' => [
+                ['$lte' => ['$expiration', $this->currentTime()]],
+                ['$eq' => ['$owner', $this->owner]],
+            ],
+        ];
+        $result = $this->collection->findOneAndUpdate(
+            ['key' => ['$eq' => $this->name]],
```

Comment:
```suggestion
            ['key' => $this->name],
```

If `name` is a string, I don't see any reason to use the verbose syntax here. According to the [server docs](https://www.mongodb.com/docs/manual/reference/operator/query/eq/#syntax), the only edge case is when the matched value is a regex.